### PR TITLE
Fixed inconcistency and bug in IUIMultilineTextInput

### PR DESCRIPTION
**Pull request by veler**
## Pull request type

Please check the type of change your PR introduces:

- [X] Bugfix
- [ ] New feature or enhancement
- [ ] UI change (please include screenshot!)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [X] Documentation content changes
- [ ] Internationalization and localization
- [ ] Other (please describe):

## What is the current behavior?

1. The method to create `IUIMultiLineTextInput` is inconsistent with `IUISingleLineTextInput`.
2. The `UIHoverTooltip` is inconsistent with `UIHighlightedTextSpan`, which they do similar things (add decoration to text)
3. There is a bug: when setting Highlights, it removed Tooltip and vice versa.

## What is the new behavior?

1. Made the API more consistent (This causes a breaking change. Will need to update NuGet package and `DevToys.Tools`)
2. Fixed the bug in `MultiLineTextInputPresenter`

## Other information

image

## Quality check

Before creating this PR:

- [X] Did you follow the code style guideline as described in [CONTRIBUTING.md](https://github.com/veler/DevToys/blob/main/CONTRIBUTING.md)
- [X] Did you build the app and test your changes?
- [ ] Did you check for accessibility? On Windows, you can use Accessibility Insights for this.
- [ ] Did you verify that the change work in Release build configuration
- [X] Did you verify that all unit tests pass
- [ ] If necessary and if possible, did you verify your changes on:
   - [X] Windows
   - [ ] macOS
   - [ ] Linux

### CODE REVIEW

File: src/app/dev/DevToys.Api/Tool/GUI/Components/IUIMultiLineTextInput.cs [hunk 1]
```diff
@@ -155,9 +155,9 @@ public static partial class GUI
     /// <remarks>
     /// This component is powered by Monaco Editor.
     /// </remarks>
-    public static IUIMultiLineTextInput MultilineTextInput()
+    public static IUIMultiLineTextInput MultiLineTextInput()
```
Comment (veler) [inline src/app/dev/DevToys.Api/Tool/GUI/Components/IUIMultiLineTextInput.cs:158]: This is to be consistent with `IUISingleLineTextInput`

File: src/app/dev/DevToys.Api/Tool/GUI/Components/UIHighlightedTextSpan.cs [hunk 2]
```diff
@@ -20,10 +20,10 @@ public UIHighlightedTextSpan(int startPosition, int length, UIHighlightedTextSpa
     /// <summary>
     /// Initializes a new instance of the <see cref="UIHighlightedTextSpan"/> class based on the original <see cref="TextSpan"/>.
     /// </summary>
-    /// <param name="original">The original <see cref="TextSpan"/>.</param>
+    /// <param name="span">The location where the text document should be highlighted <see cref="TextSpan"/>.</param>
     /// <param name="color">The color of the highlighted text span.</param>
-    protected UIHighlightedTextSpan(TextSpan original, UIHighlightedTextSpanColor color = UIHighlightedTextSpanColor.Default)
-        : base(original)
+    public UIHighlightedTextSpan(TextSpan span, UIHighlightedTextSpanColor color = UIHighlightedTextSpanColor.Default)
```
Comment (veler) [inline src/app/dev/DevToys.Api/Tool/GUI/Components/UIHighlightedTextSpan.cs:25]: To be consistent with `UIHoverTooltip`

File: src/app/dev/DevToys.Api/Tool/GUI/Components/UIHoverTooltip.cs [hunk 3]
```diff
@@ -3,24 +3,33 @@
...
 {
     /// <summary>
-    /// Contain the position of the span to search
+    /// Initializes a new instance of the <see cref="UIHoverTooltip"/> class.
     /// </summary>
-    public TextSpan Span { get; }
+    /// <param name="startPosition">The start position of the highlighted text span.</param>
+    /// <param name="length">The length of the highlighted text span.</param>
+    /// <param name="tooltip">The information to display on hover.</param>
+    public UIHoverTooltip(int startPosition, int length, string tooltip)
```
Comment (veler) [inline src/app/dev/DevToys.Api/Tool/GUI/Components/UIHoverTooltip.cs:14]: To be consistent with `UIHighlightedSpan`

File: src/app/dev/DevToys.Api/Tool/GUI/Components/UIHoverTooltip.cs [hunk 4]
```diff
@@ -3,24 +3,33 @@
 /// <summary>
 /// Represents the Tooltip to display on hover
 /// </summary>
-public record UIHoverTooltip
+public record UIHoverTooltip : TextSpan
```
Comment (veler) [inline src/app/dev/DevToys.Api/Tool/GUI/Components/UIHoverTooltip.cs:6]: To be consistent with `UIHighlightedSpan`

**Review by btiteux — APPROVED** [review]

**Pull request merged**